### PR TITLE
Endpoint for multi-files commits

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
Add endpoint for committing multiple files. The endpoint should return a response consumable by other REST endpoints, it returns JSON instead of HTML(the original endpoint response type).

### CODE REVIEW

File: routers/repo/editor.go [hunk 1]
```diff
@@ -676,6 +677,156 @@ func UploadFilePost(ctx *context.Context, form auth.UploadRepoFileForm) {
 	}
 }
 
+// UploadFilePostJson JSON response for uploading file
+func UploadFilePostJson(ctx *context.Context, form auth.UploadRepoFileForm) {
+	// `canCommit` is a poorly named; it's not responsible directly for rendering anything,
+	// it returns a boolean with commit rights
+	canCommit := renderCommitRights(ctx)
```
Comment (kasbah) [inline routers/repo/editor.go:684]: Seems like a perfect name to me? What do you mean by "responsible for rendering anything", what could it be rendering in this context?

File: routers/routes/routes.go [hunk 2]
```diff
@@ -806,6 +806,8 @@ func RegisterRoutes(m *macaron.Macaron) {
 				m.Combo("/_upload/*", repo.MustBeAbleToUpload).
 					Get(repo.UploadFile).
 					Post(bindIgnErr(auth.UploadRepoFileForm{}), repo.UploadFilePost)
+				// Same as `/_upload/*` but returns JSON
+				m.Post("/upload/*", bindIgnErr(auth.UploadRepoFileForm{}), repo.UploadFilePostJson)
```
Comment (kasbah) [inline routers/routes/routes.go:810]: Can we make this part of the API and try and upstream it to Gitea?

**Comment by AbdulrhmnGhanem** [comment]
Oach, meant 'renderCommitRights', it doesn't render anything.

**Comment by AbdulrhmnGhanem** [comment]
Yes, though not sure what signature to use for this endpoint. For single
file commit the name of the file is used as part of the url

**Comment by kasbah** [comment]
Let's merge this one in to our master and open a discussion issue on the Gitea repo.

**Pull request merged**